pkg/config/labels/instance_test.go

```diff
@@ -24,6 +24,7 @@ func TestInstance(t *testing.T) {
 	a := labels.Instance{"app": "a"}
 	a1 := labels.Instance{"app": "a", "prod": "env"}
 	a2 := labels.Instance{"app": "b", "prod": "env"}
+	b := labels.Instance{"foo": ""}
```

Comment:
I don't think this is sufficient test coverage as we don't have a test case with it negative

It would be ideal to rewrite these tests to use https://dave.cheney.net/2019/05/07/prefer-table-driven-tests format so we can have a big list of `{left, right, bool expected}` and compare all cases.

Reply:
@howardjohn Thanks for your review!
I have rewrited the tests as you addressed, PTAL :)